mlrun/model_monitoring/controller.py

```diff
@@ -301,7 +296,7 @@ def __init__(
             f"Initializing {self.__class__.__name__}", project=project
         )
 
-        self.db = mlrun.model_monitoring.get_model_endpoint_store(project=project)
+        self.db = mlrun.model_monitoring.get_store_object(project=project)
```

Comment:
As discussed try to pass this db session to the _BatchWindow & _BatchWindowGenerator instead of using get_store_object in each _BatchWindow run... I hope it wont make issues.

Reply:
Updated. It work partially because we can't pass the same db session into different processes. However, we can use a single DB session for all the required operations within a single process (e.g. get_last_analyzed + update_last_analyzed).